File: data/src/main/java/home/oleg/placesnearme/data/DaoProviderFactory.java
package home.oleg.placesnearme.data;

import android.arch.persistence.room.Room;
import android.content.Context;

import home.oleg.placesnearme.data.database.AppDatabase;
import home.oleg.placesnearme.data.database.DaoProvider;

public final class DaoProviderFactory {

    public static DaoProvider create(Context context) {
        return Room.databaseBuilder(context, AppDatabase.class, AppDatabase.NAME)
                .fallbackToDestructiveMigration()
                .build();
    }
}

File: data/src/main/java/home/oleg/placesnearme/data/DaoProviderFactory.kt
package home.oleg.placesnearme.data

import android.content.Context
import androidx.room.Room

import home.oleg.placesnearme.data.database.AppDatabase
import home.oleg.placesnearme.data.database.DaoProvider

object DaoProviderFactory {

    fun create(context: Context): DaoProvider {
        return Room.databaseBuilder(context, AppDatabase::class.java, AppDatabase.NAME)
                .fallbackToDestructiveMigration()
                .build()
    }
}
